trace/trace.go

```diff
@@ -0,0 +1,185 @@
+// Package trace provides a tracing API that in turn invokes both the `golang.org/x/net/trace` API
```

Comment:
Mostly copied from `trace` package in the Sourcegraph main repo.

Reply:
I'd be happy putting this into a new repo `github.com/sourcegraph/trace`. For example since you created this PR I updated the trace pkg in sourcegraph to be a bit better. I then improved how we log around zoekt results which would be useful to have here: https://github.com/sourcegraph/sourcegraph/pull/12554